atomic_reactor/plugins/build_orchestrate_build.py

```diff
@@ -189,13 +195,48 @@ def __init__(self, tasker, workflow, platforms, build_kwargs,
         self.build_kwargs = build_kwargs
         self.osbs_client_config = osbs_client_config
         self.config_kwargs = config_kwargs or {}
+        self.find_cluster_retry_delay = find_cluster_retry_delay
+        self.max_cluster_fails = max_cluster_fails
+        self.koji_upload_dir = self.get_koji_upload_dir()
+        self.fs_task_id = self.get_fs_task_id()
+        self.release = self.get_release()
 
         if worker_build_image:
             self.log.warning('worker_build_image is deprecated, use config_kwargs instead')
             self.config_kwargs.setdefault('build_image', worker_build_image)
 
         self.worker_builds = []
 
+    class ClusterRetryContexts(object):
+        def __init__(self, clusters, max_cluster_fails, find_cluster_retry_delay):
+            self.clusters = list(clusters)
+            self.retry_at = {}
+            self.failed = set()
+            self.fails = Counter()
+            self.max_cluster_fails = max_cluster_fails
+            self.delay = find_cluster_retry_delay
+
+        def in_retry_wait(self, cluster):
+            if cluster.name in self.retry_at:
+                return datetime.datetime.now() < self.retry_at[cluster.name]
+
+            return False
+
+        def try_again_later(self, cluster, seconds):
```

Comment:
`seconds` is unused

Reply:
changed it to use seconds.

Reply:
Why not just pass the information to this class's constructor?

Reply:
That's how I had it but I just changed it back lol